crates/server/src/authority/catalog.rs

```diff
@@ -164,6 +191,23 @@ impl Catalog {
         self.authorities.remove(name)
     }
 
+    /// Set a specified name server identifier (NSID) in responses
+    ///
+    /// The provided `NSIDPayload` will be included in responses to requests that
+    /// specify the NSID option in EDNS. Set to `None` to disable NSID.
+    ///
+    /// By default, no NSID is sent.
+    pub fn set_nsid(&mut self, payload: Option<NSIDPayload>) {
+        self.nsid_payload = payload
+    }
+
+    /// Return the name server identifier (NSID) that is used for responses (if enabled)
+    ///
+    /// See `set_nsid()` for more information.
+    pub fn nsid(&self) -> &Option<NSIDPayload> {
```

Comment:
`&Option<_>` is usually a bit of a pain, suggest an `as_ref()` call to turn this into `Option<&NSIDPayload>`.